pkg/test/expectations/expectations.go

```diff
@@ -217,6 +226,12 @@ func ExpectReconcileSucceeded(ctx context.Context, reconciler reconcile.Reconcil
 	return result
 }
 
+func ExpectReconcileFailed(ctx context.Context, reconciler reconcile.Reconciler, key client.ObjectKey) reconcile.Result {
+	result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: key})
+	Expect(err).ToNot(Succeed())
+	return result
```

Comment:
In this case we shouldn't have a result. Consider including the pretty concise result in the failure reason in the line above.